pulsar-client-cpp/lib/DefaultCAs.cc

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "DefaultCAs.h"
+
+#include <array>
+#include <iostream>
+#include <fstream>
+#include <cstdio>
+#include <boost/asio/buffer.hpp>
+
+namespace pulsar {
+
+const static std::array<std::string, 1> CAcerts = {{
+
+    // LetsEncrypt root certificate
+    // from https://letsencrypt.org/certs/lets-encrypt-x3-cross-signed.pem.txt
+    // Subject: CN=Let's Encrypt Authority X3, O=Let's Encrypt, C=US
+    //  Validity: [From: Thu Mar 17 09:40:46 PDT 2016,
+    //               To: Wed Mar 17 09:40:46 PDT 2021]
+    R"(
+-----BEGIN CERTIFICATE-----
```

Comment:
Better to use a different pem file for this and not put this info in code.

Also FWIW - I don't think we should put certificates in pulsar repo. What if due to some security vulnerability this CA cert is compromised and needs to be revoked or it expires after some time. I would prefer writing down steps/script on how the user can include letsencrypt cert in their trust store instead of adding a trusted pem file.

Reply:
The library ships as a binary `libpulsar.so`. We cannot include a PEM file in there.

> I would prefer writing down steps/script on how the user can include letsencrypt cert in their trust store instead of adding a trusted pem file.

When you install chrome/firefox you don't have to perform any operation, because the ca certs come packaged in.

Reply:
> Also FWIW - I don't think we should put certificates in pulsar repo.

This is a public certificate

>  What if due to some security vulnerability this CA cert is compromised and needs to be revoked

This is a root CA. If this is compromised, the whole internet will be in a very bad state

Reply:
> This is a public certificate
Public certs get expired or can be revoked, then it will become our headache to update it on every release or every year. If for any reason we screw up and end up pulling a compromised/hacked root cert instead - that will be really bad so we need to add a few additional steps to revalidate this root cert.

> When you install chrome/firefox you don't have to perform any operation, because the ca certs come packaged in.
What I was suggesting is to add it as a part of rpm but not as a part of the .so


If for reason you still decide to add the root certs can you make it configurable cause I am not sure if certain companies will want to add a new root cert.